lua/kznllm/specs/anthropic.lua

```diff
@@ -1,4 +1,36 @@
 local M = {}
+M.API_KEY_NAME = 'ANTHROPIC_API_KEY'
+M.URL = 'https://api.anthropic.com/v1/messages'
+
+M.MODELS = {
+  SONNET_3_5 = 'claude-3-5-sonnet-20240620',
+  OPUS_3 = 'claude-3-opus-20240229',
+  HAIKU_3 = 'claude-3-haiku-20240307',
+}
+
+M.SELECTED_MODEL = M.MODELS.SONNET_3_5
+
+M.PROMPT_TEMPLATES = {
+  --- this prompt should let the model yap into a separate buffer
+  HELPFUL_PROMPT = [[You are an AI programming assistant integrated into a code editor. Your purpose is to help the user with programming tasks as they write code.
+Key capabilities:
+- Thoroughly analyze the user's code and provide insightful suggestions for improvements related to best practices, performance, readability, and maintainability. Explain your reasoning.
+- Answer coding questions in detail, using examples from the user's own code when relevant. Break down complex topics step- Spot potential bugs and logical errors. Alert the user and suggest fixes.
+- Upon request, add helpful comments explaining complex or unclear code.
+- Suggest relevant documentation, StackOverflow answers, and other resources related to the user's code and questions.
+- Engage in back-and-forth conversations to understand the user's intent and provide the most helpful information.
+- Keep concise and use markdown.
+- When asked to create code, only generate the code. No bugs.
+- Think step by step]],
```

Comment:
prompts are actually inherently coupled to `specs` so it should make sense to leave it up to the spec entirely.

you can even say that prompts are actually coupled to models at the end of the day, which is a lot for someone just "poking around" to realize on their own.

This way I can easily bake in defaults while still allowing users to override things.